src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,116 @@
 public class StringManipulation implements StringManipulationInterface {
+    private String string;
 
-    @Override
     public String getString() {
-        return null;
+        return string;
     }
 
-    @Override
     public void setString(String string) {
+        this.string = string;
     }
 
-    @Override
+    /**
+     * Counts the number of non-empty words in the string.
+     * @return The count of words.
+     */
     public int count() {
-        return 0;
+        String[] words = string.split(" ");
+        int count = 0;
+
+        for (String word : words) {
+            if (!word.isEmpty()) {
+                count++;
+            }
+        }
+
+        return count;
```

Comment:
In your count(), I think it's smart of you to set the value to be returned (count) to 0 in case of an empty string without using a separate if statement for that.

I'm not sure if it is necessary for this assignment but one idea I could share is to use regex for the split() so that words can be separated not only by space but other types of symbols.
For example, I used an expression like this in my code:
String[] words = stringToManipulate.split("[\\s.(),:?!]+");